### DISCUSSION TITLE
Flashing GAP8 bootloader: J-Link not found in docker

### DISCUSSION THREAD

**Discussion by uldwi** [bitcraze/discussions#1498] [Q&A] [answered]
Hi,

I try to flash the gap8 firmware on the ai deck. My j-link mini is working outside the docker container but not inside it. Do you have any ideas?

I tried with the command, I found here [https://github.com/orgs/bitcraze/discussions/912](https://github.com/orgs/bitcraze/discussions/912):
`docker run --rm -it -v $PWD:/module/ -v /dev/bus/usb:/dev/bus/usb --privileged -P bitcraze/aideck:latest /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/jlink.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make clean all image flash'
`

`Error: No J-Link device found.`

Outside the docker container openocd finds the j-link. I am on Ubuntu 24.04.

**Comment by knmcguire** [comment]
If you move down the discussion thread there are some more threads to try. Perhaps there are no rights to access the jlink yet through docker.

**Comment by uldwi** [comment, marked as answer]
Thanks, I have already tried, but without success. However, I got a Olimex ARM-USB-TINY-H JTAG today and it worked out of the box.

**Reply by knmcguire** [reply]
Great! I just reopened discussion and selected your answer instead. That way it is easier for people to find support.

**Comment by Lorite** [comment]
Hi @uldwi. I was experiencing the same issue with my J-Link Mini Edu 1.00. I found out that my J-Link was in an incorrect mode. Open JLinkConfigExe and connect the board. Then, double click it. A configuration window will open. Select "SEGGER" on the last option "USB Driver (Windows)". Mine was on "WinUSB" before, which didn't work. Also, I am using Linux, not Windows, so the naming might be wrong... Feel free to check the image below. As it was initially pointed out here, the J-Link should work: https://github.com/orgs/bitcraze/discussions/912.
![Screenshot From 2025-07-02 17-20-55](https://github.com/user-attachments/assets/415f6718-8bac-4106-b1a6-0da5b3979e27)

**Reply by ToveRumar** [reply]
Hi @Lorite!
Thank you for your contribution!
